### DISCUSSION TITLE
JSON Config generator tool

### DISCUSSION THREAD

**Discussion by Bene2103** [RavenSystem/esp-homekit-devices#2012] [General]
Hello Jose,

you're doing a really really good job developing HAA, thank you very much for that effort!

I have a question about the input/output configuration.
Is it neccessary to group the i/o's into the array? Like "io":[[[2,3,4,5]6],[[7,8]2]]
Would it be possible to specify every GPIO on its own? Like "io":[[[2]6],[[3]6].......

I'm asking because i'm in the process of developing a small tool to do the JSON configuration in an graphical user interface, which would help me and maybe others to do the configuration. But right now i dont know how to group the IO's within my code.

Heres s sneak peek :-) its a pretty early stage, but i think you get the idea.
![image](https://user-images.githubusercontent.com/46553509/235665357-bfcf7830-fe35-4b03-93e2-60c5abff4b92.png)

Thank you!

**Comment by pasparakis** [comment]
Can you send details for HAA configurator?

**Comment by Bene2103** [comment]
I will give further details about my project when i feel it works good enough to share with you guys.

Right now i can generate parts from the "gerneral configuration" and thats it.

**Reply by pasparakis** [reply]
Ok..thanks

**Comment by Bene2103** [comment]
So i've made some progress, but i still dont know if the way i've done the "io" part of the GPIO configuration the right way.

But anyway, i've released a early version of my tool "HaaConfigurator" on my Github repo.

Right now the tool is only able to generate the "general" part of the JSON.

Please let me know what youre thinking! :-)

[HaaConfigurator](https://github.com/Bene2103/HaaConfigurator/releases)

**Reply by pasparakis** [reply]
![image](https://user-images.githubusercontent.com/74431015/235923888-25c6bcfe-b082-4aa8-839b-2e588a3b9abc.jpeg)

**Comment by pasparakis** [comment]
Not working...

**Comment by Bene2103** [comment]
I've tried it on a different machine and found a similar issue.

Please make sure you have the .net runtime installed, then it should run.
[.Net 7.0.203](https://download.visualstudio.microsoft.com/download/pr/60253bf7-2405-4ee4-84ea-0a5fc9764f13/f5bd3e91214abfbb89e0f522e92aca70/dotnet-sdk-7.0.203-win-x86.exe)

And if you have issues with the application, please use the issues on my github :-)

**Reply by pasparakis** [reply]
Not working 😑

**Comment by Bene2103** [comment]
So i've tried on different machines and it was working on all of them. Dont know whats the problem here.

I've released v0.15 with the ability to create accessory type 1 & 2 (switch and outlet). 
As of the generated json, i've checked a lot of possible configurations and settings and the json was always valid.

[HaaConfigurator](https://github.com/Bene2103/HaaConfigurator/releases)

**Comment by pasparakis** [comment]
And after continue I have this result ..Στάλθηκε από το iPhone μου5 Μαΐ 2023, 12:55, ο χρήστης «Bene2103 ***@***.***>» έγραψε:﻿
So i've tried on different machines and it was working on all of them. Dont know whats the problem here.
I've released v0.15 with the ability to create accessory type 1 & 2 (switch and outlet).
As of the generated json, i've checked a lot of possible configurations and settings and the json was always valid.
HaaConfigurator

**Comment by RavenSystem** [comment]
Off topic. Please, use its software GitHub repo.
